tests/components/mqtt/test_water_heater.py

```diff
@@ -1302,3 +1300,41 @@ async def test_value_template_fails(
         "TypeError: unsupported operand type(s) for *: 'NoneType' and 'int' rendering template"
         in caplog.text
     )
+
+
+@pytest.mark.parametrize(
+    "hass_config",
+    [
+        help_custom_config(
+            water_heater.DOMAIN,
+            DEFAULT_CONFIG,
+            (
+                {
+                    "temperature_unit": "F",
+                    "current_temperature_topic": "current_temperature",
+                    "temperature_high_state_topic": "temperature_high",
+                    "temperature_low_state_topic": "temperature_low",
+                },
+            ),
+        )
+    ],
+)
+async def test_target_temp_high_low_topic(
+    hass: HomeAssistant, mqtt_mock_entry: MqttMockHAClientGenerator
+) -> None:
+    """Test setting the target high and low temperature."""
```

Comment:
This test does only test setting the state after a water_heater has updated its state.
```suggestion
    """Test updating the target high and low temperature state attributes."""
```